commit 0587a0a0575c2c1e26db8fd60da9c3d26952790d (refs/changes/89/3289/5)
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2018-02-08T21:54:02+00:00 (1 year, 8 months ago)
    
    TOOLS-1957 sdcadm updateGzTools should limit the number of backups
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index 0adc4d8..72729ee 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -1523,12 +1523,34 @@ SdcAdm.prototype.updateGzTools = function updateGzTools(options, callback) {
                 common.execFilePlus({argv: argv, log: self.log}, next);
             },
 
+            // keep a finite number cn_tools backups on the usb key
+            function removeOldCNToolsTarballs(_, next) {
+                var backupPath = '/usbkey/extra/joysetup/';
+                var tarballs = fs.readdirSync(backupPath).filter(
+                    function isCNTools(p) {
+                        return (p.startsWith('cn_tools.') &&
+                                p.endsWith('tar.gz') &&
+                                p !== 'cn_tools.tar.gz');
+                    });
+                tarballs.sort();
+                tarballs.reverse();
+                var toDelete = tarballs.slice(4);
+                if (toDelete.length) {
+                    progress('Removing old cn backups: ' + toDelete.join(', '));
+                    toDelete.forEach(function rmBall(fname) {
+                        fs.unlinkSync(path.join(backupPath, fname));
+                    });
+                }
+                next();
+            },
+
             function backupCNToolsFile(_, next) {
                 if (!fs.existsSync('/usbkey/extra/joysetup/cn_tools.tar.gz')) {
                     next();
                     return;
                 }
-                var cnToolsTimestamp = new Date().toISOString();
+                var cnToolsTimestamp = new Date().toISOString()
+                    .split('.')[0].replace(/[:.-]/g, '');
                 fs.rename('/usbkey/extra/joysetup/cn_tools.tar.gz',
                       '/usbkey/extra/joysetup/cn_tools.' + cnToolsTimestamp +
                       '.tar.gz', function (err) {
